### PR TITLE
Revert "Removed GrpcClientBuilder#MultiClientBuilder (#1809)"

### PR DESCRIPTION
**Pull request by chemicL**
This reverts commit 74ac3902a0ba09c72fde79923906ad1c3ac394ee.

### CODE REVIEW

**Review by idelpivnitskiy — COMMENTED** [review]
1. Please describe in the commit message and PR description why we revert it.
2. Consider adding a test that runs a single server with 2 different services, creates 2 clients with the same transport, sends one request to each service, verifies that only one connection is created. You can use TransportObserver to validate that.

File: servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java [hunk 1]
```diff
@@ -94,4 +94,42 @@
      */
     <BlockingClient extends BlockingGrpcClient<?>> BlockingClient buildBlocking(
             GrpcClientFactory<?, BlockingClient> clientFactory);
+
+    /**
+     * Returns a {@link MultiClientBuilder} to be used to create multiple clients sharing the same underlying transport
+     * instance.
```
[the remark this answers is not in the document]
Comment (tkountis) [reply servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java:100]: Agreed with @idelpivnitskiy - or maybe specifically mention that its not meant to be used with different backends.
Also `MultiClient` is a loaded term, but I don't have a better alternative.

[the remark this answers is not in the document]
Comment (chemicL) [reply servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java:100]: We should not rename it for backwards compatibility reasons, but I agree, the name is not ideal.

File: servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java [hunk 2]
```diff
@@ -94,4 +94,42 @@
...
+     *
+     * @return A blocking <a href="https://www.grpc.io">gRPC</a> client.
```
[the remark this answers is not in the document]
Comment (chemicL) [reply servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java]: Updated.

File: servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (chemicL) [reply servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java]: I updated the javadoc.

**Comment by chemicL** [comment]
Thanks @idelpivnitskiy for the suggestion regarding a test. I added one. Will merge this PR. We can follow up if you see any further improvements are needed.

**Pull request merged**